native/src/main/java/io/ballerina/lib/sap/Client.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.lib.sap;
+
+import com.sap.conn.idoc.IDocDocumentList;
+import com.sap.conn.idoc.IDocException;
+import com.sap.conn.idoc.IDocFactory;
+import com.sap.conn.idoc.IDocRepository;
+import com.sap.conn.idoc.IDocXMLProcessor;
+import com.sap.conn.idoc.jco.JCoIDoc;
+import com.sap.conn.jco.JCoDestination;
+import com.sap.conn.jco.JCoDestinationManager;
+import com.sap.conn.jco.JCoException;
+import com.sap.conn.jco.JCoFunction;
+import com.sap.conn.jco.JCoParameterList;
+import com.sap.conn.jco.JCoRepository;
+import com.sap.conn.jco.JCoStructure;
+import io.ballerina.runtime.api.TypeTags;
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.types.RecordType;
+import io.ballerina.runtime.api.types.StructureType;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.utils.TypeUtils;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.api.values.BTypedesc;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigDecimal;
+
+public class Client {
+    private static final Logger logger = LoggerFactory.getLogger(Client.class);
+
+    public static Object initializeClient(BObject client, BMap<BString, Object> jcoDestinationConfig) {
+        try {
+            String destinationName = jcoDestinationConfig.getStringValue(StringUtils.fromString(
+                    SAPConstants.DESTINATION_ID)).getValue();
+            BallerinaDestinationDataProvider dp = new BallerinaDestinationDataProvider();
+            com.sap.conn.jco.ext.Environment.registerDestinationDataProvider(dp);
+            dp.addDestination(jcoDestinationConfig);
+            JCoDestination destination = JCoDestinationManager.getDestination(destinationName);
+            logger.debug("JCo Client initialized");
+            setDestination(client, destination);
+            return null;
```

Comment:
Do we need null return

Reply:
`initializeClient` method returns an `Object`.

Reply:
Then shouldnt we also return an object? Is that is not needed we can male the function return only error